lib/grpc/adapter/client.ex

```diff
@@ -0,0 +1,27 @@
+defmodule GRPC.ClientAdapter do
+  @moduledoc false
+
+  alias GRPC.Client.Stream
+  alias GRPC.Channel
+
+  @type opts :: any
+
+  @type channel :: Channel.t()
+  @callback connect(channel, opts) :: {:ok, channel} | {:error, any}
+  
+  @callback disconnect(channel) :: {:ok, channel} | {:error, any}
+
+  @type stream :: Stream.t()
+  @type message :: binary()
+  @callback send_request(stream, message, opts) :: stream
+
+  @type headers :: [{String.t(), String.t()}]
```

Comment:
The current implementation would need to change, as I think it currently returns a string-map